lightning/src/ln/functional_tests.rs

```diff
@@ -7380,22 +7392,46 @@ fn test_data_loss_protect() {
 	}
 
 	// Check we close channel detecting A is fallen-behind
-	nodes[1].node.handle_channel_reestablish(&nodes[0].node.get_our_node_id(), &reestablish_1[0]);
-	check_closed_event!(nodes[1], 1, ClosureReason::ProcessingError { err: "Peer attempted to reestablish channel with a very old local commitment transaction".to_string() });
-	assert_eq!(check_closed_broadcast!(nodes[1], true).unwrap().data, "Peer attempted to reestablish channel with a very old local commitment transaction");
-	check_added_monitors!(nodes[1], 1);
+	// Check if we sent the warning message when we detecting that A is fallen-behind,
+	// and we give the possibility to A to  be able to recover from error.
+	expect_warning_due_wrong_channel_reestablish!(nodes[1]);
+
+	// after the waring message sent by B, we should not able to
+	// use the channel, or reconnect with success to the channel.
+	match case {
+		DataLossProtectTestCase::UseChannel => {
+			// try to make a payment, but this should not be allowed
+			// from A -> B
+			let result = fail_to_get_route!(nodes[0], nodes[1], 1000000);
```

Comment:
Instead of a new macro and this verbose check, I think you can just do `assert!(nodes[0].node.list_usable_channels().is_empty())` or so, which is ultimately *why* the route-get fails.

Reply:
Right, less code is always better!